citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/ViafAuthorityImporter.java

```diff
@@ -94,12 +123,65 @@ public String getId() {
     }
 
     @Override
-    public AuthoritySearchResult searchAuthorities(String firstName, String lastName, int page, int pageSize) {
-        throw new UnsupportedOperationException();
+    public AuthoritySearchResult searchAuthorities(String firstName, String lastName, int page, int pageSize)
+            throws AuthorityServiceConnectionException {
+
+        pageSize = Math.min(10, pageSize);
+        String url = viafURL + viafSearchKeyword + this.getviafSearchString(firstName, lastName, page, pageSize);
+
+        ResponseEntity<ViafSearchResponseWrapper> response = null;
+        try {
+            response = restTemplate.exchange(url, HttpMethod.GET, null, ViafSearchResponseWrapper.class);
+        } catch (RestClientException ex) {
+            throw new AuthorityServiceConnectionException(ex);
+        }
+
+        List<IAuthorityEntry> authorityEntries = new ArrayList<IAuthorityEntry>();
+        AuthoritySearchResult searchResult = new AuthoritySearchResult();
+        if (response.getStatusCode() == HttpStatus.OK) {
+            ViafSearchResponse viafSearchResponse = response.getBody().getSearchRetrieveResponse();
+            if (viafSearchResponse.getRecords() != null) {
+                for (ViafSearchItem viafEntry : viafSearchResponse.getRecords()) {
+                    ViafRecordData record = viafEntry.getRecord().getRecordData();
+                    IAuthorityEntry authority = new AuthorityEntry();
+                    Iterator<Data> iterator = record.getMainHeadings().getData().iterator();
+                    if (iterator.hasNext()) {
```

Comment:
this needs a comment here, why only the first entry from the iterator is used. maybe put a short example json into the javadoc or something.

Reply:
the example should go in the javadoc along with a description of what the method is doing